### PR TITLE
No entries found cell

### LINKED ISSUE #186: "No Entries Found" state for search/filter
Current: When a user types in a search term that doesn't match any entries, they get an 'empty' state of the entry container

Expected: some sort of message stating "No entries found" to display once all entries have been filtered out.

"Learn more" link directs to: https://lockbox.firefox.com/faq.html#how-do-i-create-new-entries

## Visual Design
**InVision**
https://mozilla.invisionapp.com/share/BQEUSO0S2#/290436294_Entry_List_-_Searching_-_No_Matches

**Zeplin**
https://zpl.io/agn84N0

### PR DESCRIPTION
**Pull request by joeyg**
Resolves #186.

simulator screen shot - iphone x - 2018-06-22 at 20 50 45

### CODE REVIEW

**Comment by devinreams** [comment]
@mozilla-lockbox/product I will test this myself but can one of you take a look, too? Cool if not, just hoping to merge early this week for final testing before re-submission.

File: lockbox-ios/Storyboard/Base.lproj/ItemList.storyboard [hunk 1]
```diff
@@ -214,8 +214,59 @@
...
+                                                    <fontDescription key="fontDescription" type="system" pointSize="13"/>
+                                                    <inset key="titleEdgeInsets" minX="-15" minY="0.0" maxX="0.0" maxY="0.0"/>
+                                                    <inset key="imageEdgeInsets" minX="90" minY="0.0" maxX="0.0" maxY="0.0"/>
+                                                    <state key="normal" title="Learn more" image="arrowhead-right-16">
+                                                        <color key="titleColor" red="0.0" green="0.37647058820000001" blue="0.87450980389999999" alpha="1" colorSpace="calibratedRGB"/>
+                                                    </state>
+                                                </button>
+                                                <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="No matching entries." textAlignment="center" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="qZc-w1-SRT">
```
Comment (devinreams) [inline lockbox-ios/Storyboard/Base.lproj/ItemList.storyboard]: Can/should this be added to our Constants to keep it all in one place?

File: lockbox-ios/Storyboard/Base.lproj/ItemList.storyboard [hunk 2]
```diff
@@ -214,8 +214,59 @@
...
+                                        <rect key="frame" x="0.0" y="685" width="375" height="300"/>
+                                        <autoresizingMask key="autoresizingMask"/>
+                                        <tableViewCellContentView key="contentView" opaque="NO" clipsSubviews="YES" multipleTouchEnabled="YES" contentMode="center" preservesSuperviewLayoutMargins="YES" insetsLayoutMarginsFromSafeArea="NO" tableViewCell="k0g-dg-N0G" id="CRM-Km-Zcb">
+                                            <rect key="frame" x="0.0" y="0.0" width="375" height="300"/>
+                                            <autoresizingMask key="autoresizingMask"/>
+                                            <subviews>
+                                                <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="To see more entries here, you'll need to save them to Firefox for desktop." textAlignment="center" lineBreakMode="tailTruncation" numberOfLines="0" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="JpV-eG-grp">
```
Comment (devinreams) [inline lockbox-ios/Storyboard/Base.lproj/ItemList.storyboard]: Can/should this be added to our Constants to keep it all in one place?

File: lockbox-ios/Storyboard/Base.lproj/ItemList.storyboard [hunk 3]
```diff
@@ -214,8 +214,59 @@
...
+                                                <constraint firstItem="5gN-a0-N6V" firstAttribute="top" secondItem="JpV-eG-grp" secondAttribute="bottom" constant="15" id="eco-Sw-LZX"/>
+                                                <constraint firstItem="JpV-eG-grp" firstAttribute="centerX" secondItem="CRM-Km-Zcb" secondAttribute="centerX" id="qbU-gW-BOo"/>
+                                                <constraint firstAttribute="bottom" secondItem="5gN-a0-N6V" secondAttribute="bottom" constant="35" id="qph-D4-OMm"/>
+                                            </constraints>
+                                        </tableViewCellContentView>
+                                        <color key="backgroundColor" red="0.92941176469999998" green="0.92941176469999998" blue="0.94117647059999998" alpha="1" colorSpace="custom" customColorSpace="sRGB"/>
+                                        <connections>
+                                            <outlet property="learnMoreButton" destination="5gN-a0-N6V" id="WTm-PM-Xqd"/>
```
Comment (devinreams) [inline lockbox-ios/Storyboard/Base.lproj/ItemList.storyboard]: This learn more is intended to go to a different anchor (`#how-do-i-create-new-entries`) compared to the pre-existing `learnMoreButton` link on this view.

So... I think a different observer in the presenter is needed for this to get us to the right FAQ entry? IANAD

File: lockbox-iosTests/ItemListViewSpec.swift [hunk 4]
```diff
@@ -203,6 +203,36 @@ class ItemListViewSpec: QuickSpec {
                         expect(cell).notTo(beNil())
                     }
                 }
+
+                describe("no results palceholder") {
```
Comment (devinreams) [inline lockbox-iosTests/ItemListViewSpec.swift]: I believe this is "placeholder" typo?

**Comment by devinreams** [comment]
This "no results" state worked great for me:



Thanks again @joeyg.

File: lockbox-ios/Storyboard/Base.lproj/ItemList.storyboard (hunk 2 above)
Comment (sashei) [reply lockbox-ios/Storyboard/Base.lproj/ItemList.storyboard]: The storyboard strings aren't localized in the same way as the programmatically-included strings, it's fine to leave these as-is. Thanks, Devin!

File: lockbox-ios/Storyboard/Base.lproj/ItemList.storyboard (hunk 1 above)
Comment (sashei) [reply lockbox-ios/Storyboard/Base.lproj/ItemList.storyboard]: ^^ answered above, oops!

**Pull request merged**